In [41]:
%matplotlib inline

# Classification Iris class by using K-nearest neighbors

In [42]:
# import data from sklearn
from sklearn.datasets import load_iris
iris_datasets = load_iris()

In [43]:
print("keys of iris_datasets: \n {}".format(iris_datasets.keys()))

keys of iris_datasets: 
 dict_keys(['data', 'target', 'target_names', 'DESCR', 'feature_names'])


In [44]:
# we have three kind of Iris data
print("Target names: {}".format(iris_datasets["target_names"]))

Target names: ['setosa' 'versicolor' 'virginica']


In [45]:
# we have four features of Iris
print("Feature names: {}".format(iris_datasets["feature_names"]))

Feature names: ['sepal length (cm)', 'sepal width (cm)', 'petal length (cm)', 'petal width (cm)']


In [46]:
# we have 150 samples of Iris and four features
# 50 samples per each Iris, because we have three kind of iris
print("Shape of data: {}".format(iris_datasets["data"].shape))

Shape of data: (150, 4)


In [47]:
# data type is ndarray
print("Type of data: {}".format(type(iris_datasets["data"])))

Type of data: <class 'numpy.ndarray'>


# Estimate success degree

In [48]:
# knn is Supervised learning, so we have to split the data to train and test data
from sklearn.model_selection import train_test_split
X_train, X_test, y_train, y_test = train_test_split(iris_datasets["data"], iris_datasets["target"], random_state = 0)

In [49]:
# see splited train data
print("X_train shape: {}".format(X_train.shape))
print("y_train shape: {}".format(y_train.shape))

X_train shape: (112, 4)
y_train shape: (112,)


In [50]:
# see splited test data
print("X_test shape: {}".format(X_test.shape))
print("y_test shape: {}".format(y_test.shape))

X_test shape: (38, 4)
y_test shape: (38,)


## First model KNN

In [51]:
from sklearn.neighbors import KNeighborsClassifier
knn = KNeighborsClassifier(n_neighbors = 1)

In [52]:
# by using fit method we can make a model
knn.fit(X_train, y_train)

KNeighborsClassifier(algorithm='auto', leaf_size=30, metric='minkowski',
           metric_params=None, n_jobs=1, n_neighbors=1, p=2,
           weights='uniform')

# Evaluete model

In [53]:
y_pred = knn.predict(X_test)
print("Test set predictions: \n {}".format(y_pred))

Test set predictions: 
 [2 1 0 2 0 2 0 1 1 1 2 1 1 1 1 0 1 1 0 0 2 1 0 0 2 0 0 1 1 0 2 1 0 2 2 1 0
 2]


In [54]:
# by using score methind in knn, we can know an evaluation
print("Test set score: {:.2f}".format(knn.score(X_test, y_test)))

Test set score: 0.97


### Above score is good score to predict iris class.
#### 1 is max, it means 100 %, so 0.97 is 97 % accuracy